temporal-sdk/src/main/java/io/temporal/internal/replay/QueryResult.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.replay;
+
+import io.temporal.api.common.v1.Payloads;
+import java.util.Optional;
+
+public class QueryResult {
+  private final Optional<Payloads> responsePayloads;
+  private final boolean finalCommand;
```

Comment:
This should maybe me called `wasFinalCommand` or `finalCommandCreated` here? Because the query itself doesn't really constitute a command.

Perhaps the constructor can have a little doc explaining which situations it'd be set true in.

Reply:
Makes sense, I will rename and comment. It's true if we queried a finished execution.